packages/twenty-front/src/modules/action-menu/actions/utils/getActionConfig.ts

```diff
@@ -2,26 +2,81 @@ import { DEFAULT_RECORD_ACTIONS_CONFIG } from '@/action-menu/actions/record-acti
 import { WORKFLOW_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowActionsConfig';
 import { WORKFLOW_RUNS_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowRunsActionsConfig';
 import { WORKFLOW_VERSIONS_ACTIONS_CONFIG } from '@/action-menu/actions/record-actions/constants/WorkflowVersionsActionsConfig';
+import { NoSelectionRecordActionKeys } from '@/action-menu/actions/record-actions/no-selection/types/NoSelectionRecordActionsKeys';
 import { ActionConfig } from '@/action-menu/actions/types/ActionConfig';
 import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
 import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
 import { isDefined } from 'twenty-shared/utils';
 
-export const getActionConfig = (
-  objectMetadataItem?: ObjectMetadataItem,
-): Record<string, ActionConfig> => {
+export const getActionConfig = ({
+  objectMetadataItem,
+  notePermissions,
+  taskPermissions,
+  companyPermissions,
+  personPermissions,
+  opportunityPermissions,
+  workflowPermissions,
+}: {
+  objectMetadataItem?: ObjectMetadataItem;
+  notePermissions: { canRead: boolean };
+  taskPermissions: { canRead: boolean };
+  companyPermissions: { canRead: boolean };
+  personPermissions: { canRead: boolean };
+  opportunityPermissions: { canRead: boolean };
+  workflowPermissions: { canRead: boolean };
+}) => {
   if (!isDefined(objectMetadataItem)) {
     return {};
   }
 
+  let actionsConfig: Record<string, ActionConfig>;
+
   switch (objectMetadataItem.nameSingular) {
-    case CoreObjectNameSingular.Workflow:
-      return WORKFLOW_ACTIONS_CONFIG;
-    case CoreObjectNameSingular.WorkflowVersion:
-      return WORKFLOW_VERSIONS_ACTIONS_CONFIG;
-    case CoreObjectNameSingular.WorkflowRun:
-      return WORKFLOW_RUNS_ACTIONS_CONFIG;
+    case CoreObjectNameSingular.Workflow: {
+      actionsConfig = WORKFLOW_ACTIONS_CONFIG;
+      break;
+    }
+    case CoreObjectNameSingular.WorkflowVersion: {
+      actionsConfig = WORKFLOW_VERSIONS_ACTIONS_CONFIG;
+      break;
+    }
+    case CoreObjectNameSingular.WorkflowRun: {
+      actionsConfig = WORKFLOW_RUNS_ACTIONS_CONFIG;
+      break;
+    }
     default:
-      return DEFAULT_RECORD_ACTIONS_CONFIG;
+      actionsConfig = DEFAULT_RECORD_ACTIONS_CONFIG;
+  }
+
+  const filteredActionsConfig = { ...actionsConfig };
+
+  if (!notePermissions.canRead) {
+    delete filteredActionsConfig[NoSelectionRecordActionKeys.GO_TO_NOTES];
```

Comment:
we should avoid side effects (deleting from existing array). We prefer a .filter pattern